### PR TITLE
Spark plugin docs

### PR DESCRIPTION
**Pull request by kumare3**
- Explains how to install the plugin
- How to enable it

### CODE REVIEW

File: rsts/plugins/spark_k8s.rst [hunk 1]
```diff
@@ -3,3 +3,86 @@
...
+
+Cons:
+-----
+#. Short running, bursty jobs are not a great fit - because of the container overhead
+#. No interactive spark capabilities available with Flyte K8s spark. It is useful for running, adhoc and/or scheduled jobs
+
+
+How to enable Spark in flyte backend?
+======================================
+Flyte Spark uses `Spark On  K8s Operator <https://github.com/GoogleCloudPlatform/spark-on-k8s-operator>`_ and a custom built `Flyte Spark Plugin <https://pkg.go.dev/github.com/lyft/flyteplugins@v0.5.25/go/tasks/plugins/k8s/spark>`_.
```
Comment (katrogan) [inline rsts/plugins/spark_k8s.rst:39]:
```suggestion
Flyte Spark uses the `Spark On K8s Operator <https://github.com/GoogleCloudPlatform/spark-on-k8s-operator>`_ and a custom built `Flyte Spark Plugin <https://pkg.go.dev/github.com/lyft/flyteplugins@v0.5.25/go/tasks/plugins/k8s/spark>`_.
```

**Review by wild-endeavor — APPROVED** [review]

File: rsts/plugins/spark_k8s.rst [hunk 2]
```diff
@@ -3,3 +3,86 @@
...
+        @task(
+            task_config=Spark(
+                # this configuration is applied to the spark cluster
+                spark_conf={
+                    "spark.driver.memory": "1000M",
+                    "spark.executor.instances": "2",
+                    "spark.driver.cores": "1",
+                }
+            ),
+            cache_version="1",
```
Comment (katrogan) [inline rsts/plugins/spark_k8s.rst:72]: if we're using caching in this example don't you need `cache=True` here too?

**Review by katrogan — APPROVED** [review]

File: rsts/plugins/spark_k8s.rst (hunk 2 above)
Comment (kumare3) [reply rsts/plugins/spark_k8s.rst:72]: yes

File: rsts/plugins/spark_k8s.rst [hunk 3]
```diff
@@ -3,3 +3,86 @@
...
```
Comment (kumare3) [inline rsts/plugins/spark_k8s.rst:72]:
```suggestion
            cache_version="1",
            cache=True,
```

**Review by katrogan — APPROVED** [review]

**Pull request merged**